### PR TITLE
Map visualization for multiple devices

### PR DESCRIPTION
**Pull request by cfrancisco**
Initially, was added a devices visualization based on example created by @mmagr. 
Besides, was created custom pins (to be used in markers) and added a context menu, shown when user right-click on a marker.

### CODE REVIEW

File: src/js/views/devices/DeviceMap.js [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+var blackPin = L.divIcon({className: 'icon-marker bg-black'});
+var redPin = L.divIcon({className: 'icon-marker bg-red'});
+
+
+
+class PositionRenderer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      zoom: 13,
```
Comment (mmagr) [inline src/js/views/devices/DeviceMap.js]: unless we're reading this from somewhere else (i.e. like storing that on the session) this should be removed, right?

File: src/js/views/devices/DeviceMap.js [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+                <div className="contextMenu--separator" />
+                <Link to={"/device/id/" + this.state.selected_device_id + "/detail"} title="View details">
+                <div className="contextMenu--option"><i className="fa fa-fa-info-circle" />Details</div>
+                </Link>
+                <div className="contextMenu--option"  onClick={this.handleTracking}><i className="fa fa-compass" />Tracking</div>
+            </div>
+          ) : null
+
+    return (
+      <Map center={parsedEntries[0].pos} zoom={13} ref={m => {this.leafletMap = m;}}   onContextMenu={this._handleClick}  onMoveStart={this._handleMoveStart} >
```
Comment (mmagr) [inline src/js/views/devices/DeviceMap.js]: zoom should use use the value from this component's state, right?
Also, do check the note on the need of zoom in this component's state.

File: src/js/views/devices/DeviceMap.js [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+        {contextMenu}
```
Comment (mmagr) [inline src/js/views/devices/DeviceMap.js]: This is actually just a popup, but triggered on right click. Is that correct?
If so, why not just change to left click and use popups?

File: src/js/views/devices/DeviceMap.js (hunk 1 above)
Comment (cfrancisco) [reply src/js/views/devices/DeviceMap.js]: yes!

File: src/js/views/devices/DeviceMap.js (hunk 3 above)
Comment (cfrancisco) [reply src/js/views/devices/DeviceMap.js]: Well, I was thinking as a real context menu, not a popup, but we can change its behavior.

File: src/js/views/devices/DeviceMap.js [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+        <ReactResizeDetector handleWidth onResize={this.resize.bind(this)} />
+        <TileLayer
+          url='http://{s}.tile.osm.org/{z}/{x}/{y}.png'
+          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+        />
+        {parsedEntries.map((k) => (
+          <Marker id={"dev_"+k.id} onContextMenu={this._handleContextMenu} position={k.pos} key={k.id} icon={k.pin}  >
```
[the remark this answers is not in the document]
Comment (cfrancisco) [reply src/js/views/devices/DeviceMap.js]: Nice! I will try it!

**Pull request merged**